plugins/chat/assets/javascripts/discourse/components/mention-warnings.js

```diff
@@ -0,0 +1,142 @@
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import { escapeExpression } from "discourse/lib/utilities";
+import { htmlSafe } from "@ember/template";
+import { inject as service } from "@ember/service";
+
+export default class MentionWarnings extends Component {
+  @service siteSettings;
+  @service currentUser;
+  tagName = "";
+
+  get show() {
+    return (
+      this.args.tooManyMentions ||
+      this.args.unreachableGroupMentions.length > 0 ||
+      this.args.overMembersLimitGroupMentions.length > 0
+    );
+  }
+
+  get listStyleClass() {
+    if (this.args.tooManyMentions) {
+      return "chat-mention-warnings-list-simple";
+    }
+
+    if (
+      this.args.unreachableGroupMentions.length +
+        this.args.overMembersLimitGroupMentions.length >
+      1
+    ) {
+      return "chat-mention-warnings-list-multiple";
+    } else {
+      return "chat-mention-warnings-list-simple";
+    }
+  }
+
+  get warningHeaderText() {
+    const errorsCount =
+      this.args.unreachableGroupMentions.length +
+      this.args.overMembersLimitGroupMentions.length;
+    if (this.args.mentionsCount <= errorsCount || this.args.tooManyMentions) {
+      return I18n.t("chat.mention_warning.groups.header.all");
+    } else {
+      return I18n.t("chat.mention_warning.groups.header.some");
+    }
+  }
+
+  get tooManyMentionsBody() {
+    if (!this.args.tooManyMentions) {
+      return;
+    }
+
+    let notificationLimit = escapeExpression(
+      I18n.t("chat.mention_warning.groups.notification_limit")
+    );
+
+    if (this.currentUser.staff) {
+      notificationLimit = htmlSafe(
+        `<a 
```

Comment:
Im not super fan of building HTML like this in the compobent

Reply:
Hmm it's tricky because I need to pass it to another translation as an argument. Is there a better way?